torch/ao/quantization/fuser_method_mappings.py

```diff
@@ -226,6 +227,37 @@ def reversed(is_qat, x, w):
     (nn.BatchNorm3d, nn.ConvTranspose3d): reverse2(fuse_convtranspose_bn),
 }
 
+def get_valid_patterns(op_pattern):
```

Comment:
could we explain why this is needed? seems like the only place this is used is for getting a fuser method, what's the use case for getting a fuser method where one of the nodes is a wildcard?

Reply:
this is because the node that is supposed to be matched with MatchAllNode could be some normal node, for example, nn.MaxPool, we have this example in the test case, we are trying to match
(torch.add, MatchAllNode, (torch.nn.ReLU, torch.nn.Conv2d)) and the actual pattern we have is:
(torch.add, torch.nn.MaxPool, (torch.nn.ReLU, torch.nn.Conv2d)) --> here torch.nn.MaxPool actually can match MatchAllNode, so we need to generate this key in fuser_method lookup (fuser_method_mapping maps from a pattern to the fuser_method corresponding to it.

Alternatively we could also change the way we do lookup I think, in that case, we need to go through all patterns in the fuser_method_mapping one by one (this makes sense as well I think)

Reply:
`(torch.add, MatchAllNode, (torch.nn.ReLU, torch.nn.Conv2d))`

is the fusion going to be implemented with wildcard support?  Some kind of backend specific fusion?  Is there some example of how this works in a real backend?  I think I'm just confused on how can a backend fuse a wildcard node without knowing the node type.

Reply:
we have an example in `test_fuse_custom_pattern`. 
re "how can a backend fuse a wildcard node without knowing the node type." - it's just for input I think, `(torch.add, MatchAllNode, (torch.nn.ReLU, torch.nn.Conv2d))` just means the fusion can work no matter what node the first input of add is, actually this is True for all patterns, maybe to make everything consistent, this pattern should be:

```
(torch.add, MatchAllNode, (torch.nn.ReLU, (torch.nn.Conv2d, MatchAllNode)))
```
since we don't care about the type for input of torch.nn.Conv2d either

Reply:
hmm, should we have a distinction between "match any input" versus "match any node" in the matching specification?  I think I understand the use case now, but the pattern looks like it's actually matching a node and not the input.

Reply:
this is something inherent in fx node.. fx node is both a Node and the output Value. "input" is a concept of Value, not Node, but we don't have such thing in FX as a separate concept right now I think.
I feel that to be consistent, this pattern is probably better than the current pattern though: 
```
(torch.add, MatchAllNode, (torch.nn.ReLU, (torch.nn.Conv2d, MatchAllNode)))
```

Reply:
To respond to the question directly, the difference between "match any input" and "match any node" would be the first one doesn't include the input in the pattern and the second does? I feel currently we doesn't seem to have use cases for "match any node", it's hard to imagine how a pattern could include arbitrary node, so we probably won't have that use case?

I can think of use cases like, match some elementwise ops, or match some reduction ops, but for these we can just spell out all the patterns for now I think.

Reply:
Sure, it makes sense after your comments.  Maybe we can use naming or documentation in the future to make this clearer.